internal/bundlereader/loaddirectory_test.go

```diff
@@ -29,7 +29,29 @@ func TestGetContent(t *testing.T) {
 		name               string
 		directoryStructure fsNode
 		expectedFiles      map[string][]byte
+		source             string
+		auth               bundlereader.Auth
+		expectedErr        string
 	}{
+		{
+			name: "ensure panic doesn't occur when InsecureSkipVerify is set to false (#3782)",
+			directoryStructure: fsNode{
+				name: "fleet.yaml",
+				contents: `namespace: fleet-helm-oci-with-auth-example
+		 helm:
+		   chart: "oci://ghcr.io/fleetqa/fleet-qa-examples/fleet-test-configmap-chart"
+		   version: "0.1.0"
+		   values:
+		     replicas: 2`,
+			},
+			source: "oci://ghcr.io/fleetqa/fleet-qa-examples/fleet-test-configmap-chart",
+			auth: bundlereader.Auth{
+				Username: "foo",
+				Password: "bar",
+				//InsecureSkipVerify: true, // When commented out, the panic disappears
+			},
+			expectedErr: "response status code 403: denied",
```

Comment:
Following my comment above on OCI URLs, this could become something like:
```suggestion
			expectedErr: "no such host",
```